Kueue: Add customconfigs e2e test jobs

kaisoz opened the pull request:
This PR adds CI jobs for running Kueue customconfigs e2e jobs both at PR time and periodically. This is needed as follow up to https://github.com/kubernetes-sigs/kueue/pull/4112

mimowo commented:
/assgin 
cc @tenzen-y 
as discussed with @kaisoz offline it is hard to see what is changed in the copy-pasted configs with adjustments, so I asked to either:
1. describe what is changed after copy and which config served as source;
2. produce sharable diff using https://www.diffchecker.com/

Either is fine or both

kaisoz commented:
@mimowo @tenzen-y 

Here are the diffs for the copy-pasted content in `diffchecker`:

* [kueue-presubmits-main.yaml](https://www.diffchecker.com/s2jtbAJH/)
* [kueue-release-blocking-main.yaml](https://www.diffchecker.com/gc2owXYZ/)

In summary, for each one, I copied the `test-tas-e2e` configs and
* renamed the job
* renamed the `testgrid-tab-name`
* adjusted the job description
* replaced the make target

mimowo commented:
/lgtm
/approve
Since you follow the current naming convention moving it forward.

However, as a follow up I'm considering to rename the e2e CI Jobs to all start with `pull-kueue-test-e2e`, then they would nicely group together, wdyt  @tenzen-y? OTOH then we may lose the accumulated history of job results if we also change in `testgrid-tab-name`.

tenzen-y commented:
@kaisoz Would you mind to open PR?

kaisoz commented:
> @kaisoz Would you mind to open PR?
> 

Sure! I'll do it shortly 😊
